File: Server/DataTypes/Squad.cs
using Fps.Library.DataTypes.BaseDataTypes.Attributes;
using Fps.Library.DataTypes.BaseDataTypes.BaseDataType;
using IFK.Server.DataTypes.Issues;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace IFK.Server.DataTypes
{
    public class Squad : BaseDataType
    {
        public string Key { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        [UpdateIgnore]
        public virtual List<User> Composition { get; set; }
        [UpdateIgnore]
        public virtual List<Issue> Issues { get; set; }
        [NotMapped]
        public IEnumerable<Issue> SprintIssues
        {
            get
            {
                return Issues.Where(issue => Status.Backlog != issue.Status && Status.Closed != issue.Status);
            }
        }
        public virtual Sprint Sprint { get; set; }
        //public virtual List<Notification> Notifications { get; set; } 

        public Squad()
        {
            Key = string.Empty;
            Name = string.Empty;
            Description = string.Empty;
            Composition = new List<User>();
            Issues = new List<Issue>();
            Sprint = null;
            //Notifications = new List<Notification>();
        }
    }
}

File: Server/Content/Scripts/DataTypes/Squad.ts
import { BaseDataType } from './BaseDataType';
import { Issue } from './Issues/Issue';
import { Notification } from './Notifications/Notification';
import { Sprint } from './Sprint';
import { User } from './User';

export class Squad extends BaseDataType
{
    public Key: string;
    public Name: string;
    public Description: string;

    public Composition: Array<User>;
    public Issues: Array<Issue>;

    public constructor()
    {
        super();
        this.Key = "";
        this.Name = "";
        this.Description = "";

        this.Composition = new Array<User>();
        this.Issues = new Array<Issue>();
    }
}